sympy/utilities/lambdify.py

```diff
@@ -903,6 +905,97 @@ def _module_present(modname, modlist):
             return True
     return False
 
+class _RedundancySliter:
```

Comment:
Why is this a class rather than just some functions?

Reply:
I am not sure there are good reasons for this. I needed to do recursive and have 'global' variables. Making a class makes it possible to implement this in a clear way. I am not able to implement this "clean" with a single function. But it is undoubtedly possible. How is it embarrassing to have a class?

Reply:
I'm not sure what you mean by "global" variables.

I don't find the class to be very clear compared to just some functions e.g.:
```python
def count(expr):
    return 1 + sum(count(child) for child in expr.args)
```
There is already a function for this by the way:
```python
In [1]: e = 1 + x

In [2]: e
Out[2]: x + 1

In [3]: e.count
Out[3]: <bound method Basic.count of x + 1>

In [4]: e.count(Expr)
Out[4]: 3
```

Reply:
Ha okay, I understood to put everything in a single function! If you prefer to do multiple functions rather than one object, I have nothing against it. Will you be responsible for making the modification or will I take care of it?

Reply:
It's for you to make the changes. You can just commit and push to the same branch to update the PR.

There are also test failures that need to be addressed.